prow/cmd/deck/template/spyglass.html

```diff
@@ -47,3 +49,7 @@
 {{end}}
 
 {{template "page" (settings mobileUnfriendly darkMode "spyglass" .)}}
+
+<div id="rerun">
+  <div id="rerun-content"></div>
```

Comment:
Rather than always adding a rerun element at the bottom of the page, please conditionally add the rerun element as part of the link header defined on lines 23-35. We should make the rerun link more visible and only show up if there is still a ProwJob resource available to rerun. Notice that spyglass can return an empty string ProwJob name without erroring: https://github.com/kubernetes/test-infra/blob/f7e21a3c18f4f4bbc7ee170675ed53e4544a0632/prow/spyglass/spyglass.go#L298-L299
This is because spyglass can display job results from GCS even after the ProwJob k8s resource has been garbage collected. In this case we can't rerun since we don't have a ProwJob resource. (Usually a copy of the ProwJob resource is stored with the job results in GCS so we could actually use that to rerun, but if we want to add that lets do it in a separate PR.)

Reply:
Please let me know if I understood it right before fixing it:
1. Move the rerun button icon from the top of the page (next to the title) to be part of the links header.
I tried that at first with the icon, but it looked bad, so I thought it would look nicer at the top. Do you prefer only a "Rerun" title?
2. Make the rerun link appear conditionally in case the ProwJob resource is available.
I used the same implementation as the deck view. It shows the button and the popup with the command copy button unconditionally but there's an additional **Rerun** button that appears only in case _rerunCreatesJob_ boolean option is true.
So, do you want the button link not to be shown at all or did you refer only to the internal rerun button? Do you want the same behavior to be aligned with the deck view or does spyglass is expected to behave differently?

Reply:
1. I didn't realize the icon is currently near the title, that position is fine. I thought the button would appear at the bottom of the page because it is not defined in the `title` definition on line 1. Sounds like that is not the case for some reason?
2. Spyglass pages have an important distinction from the main deck page: The main page is entirely based off of ProwJob resource data. This data is garbage collected after a short amount of time (~24 hours on high traffic instances). So the main page only shows jobs that have run recently enough to still have ProwJob resources in the cluster. Spyglass pages are different in that they also consume job result data from GCS which persists for months or years. Since the ProwJob resource is needed to rerun the job, spyglass pages need to only show the rerun button when the ProwJob resource is still available in the cluster (or otherwise disable the button).  ProwJob resources are typically uploaded to GCS as json so we could make this smarter in a follow up PR, but we'll still need a way to disable or hide the button since we can't assume the ProwJob will be uploaded to GCS.
So with respect to whether the button link or internal rerun button should be hidden: both since neither can be provided if a ProwJob resource is not available.

Reply:
1. Yes. The icon is added as part of the load function.
https://github.com/kubernetes/test-infra/blob/953aa5b87533c2294f32d86159918eb7947cc767/prow/cmd/deck/static/spyglass/spyglass.ts#L185-L189
The div at the bottom of the page is the box that's going to pop up when the icon is clicked. Its style dimensions are defined on the CSS
https://github.com/kubernetes/test-infra/blob/779dbad39c4856489a1efe7b2485cfe9c42dd430/prow/cmd/deck/static/style.css#L308-L330
2. I made the button appear only in case prowJobName is avaiable.

Reply:
:+1: LGTM